dependencies/FindO2GPU.cmake

```diff
@@ -200,20 +230,27 @@ if(ENABLE_HIP)
     set_package_properties(rocprim PROPERTIES TYPE REQUIRED)
     set_package_properties(rocthrust PROPERTIES TYPE REQUIRED)
   endif()
+  if (CMAKE_HIP_COMPILER)
+    enable_language(HIP)
+    message(STATUS "HIP language enabled: ${CMAKE_HIP_COMPILER}")
+  endif()
   if(hip_FOUND AND hipcub_FOUND AND rocthrust_FOUND AND rocprim_FOUND AND hip_HIPCC_EXECUTABLE AND hip_HIPIFY_PERL_EXECUTABLE)
     set(HIP_ENABLED ON)
     set_target_properties(roc::rocthrust PROPERTIES IMPORTED_GLOBAL TRUE)
-    message(STATUS "HIP Found (${hip_HIPCC_EXECUTABLE} version ${hip_VERSION})")
+    message(STATUS "HIP Found (${hip_HIPCC_EXECUTABLE} version ${hip_VERSION})") # not needed anymore eventually if we migrate to native HIP clang++
     set(O2_HIP_CMAKE_CXX_FLAGS "-fgpu-defer-diag -mllvm -amdgpu-enable-lower-module-lds=false -Wno-invalid-command-line-argument -Wno-unused-command-line-argument -Wno-invalid-constexpr -Wno-ignored-optimization-argument -Wno-unused-private-field -Wno-pass-failed")
```

Comment:
Can't we put the O2_HIP_... flags now into CMAKE_HIP_FLAGS?
Though not so sure about the linker flags.

Reply:
that's the plan, during the process I make them to coexist until everything is ported.